### ISSUE TITLE
CMake could probably auto-detect libiomp5 using LD_LIBRARY_PATH

### ISSUE THREAD

**Issue by TiborGY** [psi4/psi4#2806]
When doing condaless builds, using the Numpy/MKL supplied by the OneAPI toolkits, the CMake system can find MKL but fail to find libiomp5. This happens if Intel's environment setup script has been run (`source /opt/intel/oneapi/setvars.sh`).
The [CMake docs suggest](https://cmake.org/cmake/help/latest/module/FindBLAS.html) also doing `. /opt/intel/oneapi/compiler/latest/env/vars.sh` but that improved nothing for me.

The only thing that worked is figuring out exactly where libiomp5.a is (which turned out not to be easily googleable!), and setting `-DOpenMP_LIBRARY_DIRS=/opt/intel/oneapi/compiler/2022.2.1/linux/compiler/lib/intel64` for CMake.

If CMake could locate libiomp5 after the Intel script has set things up, this would be a far less frustrating experience. And I think it should have a good shot at automatically doing this, under many circumstances, as the Intel script populates `LD_LIBRARY_PATH` with a list of directories, one of which is almost certain to contain libiomp5.a/.so.

I am not enough of a build system wizard to do this, but CMake could theoretically try looking for libiomp5.a in every directory listed in `LD_LIBRARY_PATH` and only require the use of `-DOpenMP_LIBRARY_DIRS=` if none of them contain it.

**Comment by loriab** [comment]
Yeah, the BLAS+OpenMP detection in Psi4 much predates the CMake FindBLAS, OneAPI, and Intel MKLConfig.cmake reforms. I'm not keen on any immediate changes before the upcoming release, but I agree that it's worth revisiting to see if some of the new tools streamline the build or solve more situations w/o intervention. (Though I personally don't like having environment variables influence CMake. :-) )

Btw iirc, OpenMP doesn't like to be statically linked, so you might favor `libiomp5.so`.
